scripts/foundry/contract-upgrades/v3.1.0/README.md

```diff
@@ -0,0 +1,118 @@
+# Nitro contracts 3.1.0 upgrade
+
+To upgrade to Nitro contracts 3.1.0 (BoLD), you will need to use the [BOLDUpgradeAction](https://github.com/EspressoSystems/nitro-contracts/blob/main/src/rollup/BOLDUpgradeAction.sol) from the [nitro-contract](https://github.com/EspressoSystems/nitro-contracts) repo.
+
+BOLDUpgradeAction will perform the following actions:
+
+1. Upgrade bridge, inbox, rollupEventInbox, outbox, sequencerInbox to v3.1.0
+2. Deploy new v3.1.0 BoLD challenge manager
+3. Migrate v2 rollup into a new v3.1.0 rollup address
+4. Setup the rollup according to the new config and use the latest confirmed assertion on the old rollup as genesis of the new rollup
+
+Note that contracts without code changes are not upgraded. It is normal to have some contracts still in the old version after the upgrade as they are equivalent to the new version. There are no associated ArbOS upgrade for this version.
+
+## Requirements
+
+**Nitro contracts**: This upgrade only support upgrading from the v2.1.3 version of Espresso nitro-contracts.
+
+## How to use it
+
+1. Clone nitro-contract repo
+
+```
+$ git clone https://github.com/EspressoSystems/nitro-contracts.git
+$ cd nitro-contracts
+```
+
+2. Checkout the v3.1.0-c6f2078 tag
+
+```
+$ git checkout v3.1.0-c6f2078
```

Comment:
Should we specify that this is for vanilla nitro?

Reply:
Any non integration migration docs will go to celestia integration branch so here is implicitly implied

Reply:
True that makes sense